docs/core/resilience/http-resilience.md

```diff
@@ -0,0 +1,215 @@
+| **5** | Attempt timeout (per endpoint) | The attempt timeout pipeline limits each request attempt duration and throws if its exceeded. |
+
+### Customize resilience handlers
+
+For finite control, you can customize the resilience handlers by calling the `AddResilienceHandler` extension method. This method takes a delegate that configures the `ResiliencePipelineBuilder<HttpResponseMessage>` instance that is used to create the resilience strategies.
+
+To configure a named resilience handler, call the `AddResilienceHandler` extension method with the name of the handler. The following example configures a named resilience handler called `CustomResilienceHandler`.
+
+```csharp
+builder.AddResilienceHandler("CustomPipeline", static builder =>
+{
+    // See: https://www.pollydocs.org/strategies/retry.html
+    builder.AddRetry(new RetryStrategyOptions<HttpResponseMessage>
+    {
+        BackoffType = DelayBackoffType.Exponential,
+        MaxRetryAttempts = 5,
+        UseJitter = true
+    });
+
+    // See: https://www.pollydocs.org/strategies/circuit-breaker.html
+    var options = new CircuitBreakerStrategyOptions<HttpResponseMessage>()
+    {
+        SamplingDuration = TimeSpan.FromSeconds(10),
+        FailureRatio = 0.2,
+        MinimumThroughput = 3,
+        ShouldHandle = static args =>
+        {
+            // Handles HTTP status codes 408 and 429.
+            return ValueTask.FromResult(args is
+            {
+                Outcome.Result.StatusCode:
+                    HttpStatusCode.RequestTimeout or HttpStatusCode.TooManyRequests
+            });
+        }
+    };
+    builder.AddCircuitBreaker(options);
+
+    // See: https://www.pollydocs.org/strategies/timeout.html
+    builder.AddTimeout(TimeSpan.FromSeconds(5));
+});
+```
+
+The preceding code:
+
+- Adds a resilience handler with the `"CustomPipeline"` as the `pipelineName` to the service container.
+- Adds a retry strategy with exponential backoff, five retries, and jitter preference to the resilience builder.
+- Adds a circuit breaker strategy with a sampling duration of 10 seconds, a failure ratio of 0.2 (20%), a minimum throughput of three, and a predicate that handles `RequestTimeout` and `TooManyRequests` status codes to the resilience builder.
+- Adds a timeout strategy with a timeout of five seconds to the resilience builder.
+
+There are many options available for each of the resilience strategies. For more information, see the [Polly docs: Strategies](https://www.pollydocs.org/strategies).
+
+## Example usage
+
+You app relies on [dependency injection](../extensions/dependency-injection.md) to resolve the `ExampleClient` and its corresponding <xref:System.Net.Http.HttpClient>, but for this example, since the <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> was manually instantiated this code builds the <xref:System.IServiceProvider> and resolves the `ExampleClient` from it.
```

Comment:
```suggestion
You app relies on [dependency injection](../extensions/dependency-injection.md) to resolve the `ExampleClient` and its corresponding <xref:System.Net.Http.HttpClient> but, for this example, since the <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> was manually instantiated, this code builds the <xref:System.IServiceProvider> and resolves the `ExampleClient` from it.
```